great_expectations/rule_based_profiler/domain_builder/multi_column_domain_builder.py

```diff
@@ -0,0 +1,79 @@
+from typing import List, Optional, Union
+
+from great_expectations.core.batch import Batch, BatchRequest, RuntimeBatchRequest
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.domain_builder import ColumnDomainBuilder
+from great_expectations.rule_based_profiler.types import Domain, ParameterContainer
+
+
+class MultiColumnDomainBuilder(ColumnDomainBuilder):
+    """
+    This DomainBuilder uses relative tolerance of specified map metric to identify domains.
+    """
+
+    def __init__(
+        self,
+        include_column_names: Optional[Union[str, Optional[List[str]]]] = None,
+        batch_list: Optional[List[Batch]] = None,
+        batch_request: Optional[
+            Union[str, BatchRequest, RuntimeBatchRequest, dict]
+        ] = None,
+        data_context: Optional["DataContext"] = None,  # noqa: F821
+    ):
+        """
+        Args:
+            include_column_names: Explicitly specified desired columns (if None, it is computed based on active Batch).
+            batch_list: explicitly specified Batch objects for use in DomainBuilder
+            batch_request: BatchRequest to be optionally used to define batches to consider for this domain builder.
+            data_context: DataContext associated with this profiler.
+        """
+        super().__init__(
+            include_column_names=include_column_names,
+            exclude_column_names=None,
+            include_column_name_suffixes=None,
+            exclude_column_name_suffixes=None,
+            semantic_type_filter_module_name=None,
+            semantic_type_filter_class_name=None,
+            include_semantic_types=None,
+            exclude_semantic_types=None,
+            batch_list=batch_list,
+            batch_request=batch_request,
+            data_context=data_context,
+        )
+
+    @property
+    def domain_type(self) -> Union[str, MetricDomainTypes]:
+        return MetricDomainTypes.MULTICOLUMN
```

Comment:
I know we accept string representations but is this not a single return type of `MetricDomainTypes`?

Reply:
@cdkini Just adhering the the interface.  If you think we should change the interface, let us discuss -- if no implications, fine with me (but not sure right now).  Thanks!